devicetypes/smartthings/tint-remote-controller.src/tint-remote-controller.groovy

```diff
@@ -0,0 +1,308 @@
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.lights.philips.hue-single", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.lights.philips.hue-single", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.lights.philips.hue-single", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.lights.philips.hue-single", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+			tileAttribute ("device.color", key: "COLOR_CONTROL") {
+				attributeState "color", action:"color control.setColor"
+			}
+		}
+		controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range:"(2700..6500)") {
+			state "colorTemperature", action:"color temperature.setColorTemperature"
+		}
+
+		main(["switch"])
+		details(["switch", "colorTempSliderControl"])
+	}
+}
+
+private getGROUPS_CLUSTER() { 0x0004 }
+private getADD_GROUP_COMMAND() { 0x00 }
+private getMOVE_TO_COLOR_COMMAND() { 0x07 }
+private getON_OFF_ATTRIBUTE() { 0x0000 }
+private getMOVE_TO_COLOR_TEMPERATURE_COMMAND() { 0x0A }
+private getSTEP_COMMAND() { 0x02 }
+private getWRITE_ATTRIBUTES_COMMAND() { 0x02 }
+
+def parse(String description) {
+	def event = zigbee.getEvent(description)
+	def result = []
+	if (event) {
+		log.debug "Creating event: ${event}"
+		result = createEvent(event)
+	} else {
+		if (isAttrOrCmdMessage(description)) {
+			result = parseAttrCmdMessage(description)
+		} else {
+			log.warn "Unhandled message came in"
+		}
+	}
+	log.debug "Parse returned: ${result}"
+	return result
+}
+
+private parseAttrCmdMessage(description) {
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	if (descMap.clusterInt == zigbee.ONOFF_CLUSTER) {
+		getOnOffEvent()
+	} else if (descMap.clusterInt == zigbee.COLOR_CONTROL_CLUSTER && descMap.commandInt == MOVE_TO_COLOR_COMMAND) {
+		getColorEvent(descMap)
+	} else if (descMap.clusterInt == zigbee.COLOR_CONTROL_CLUSTER && descMap.commandInt == MOVE_TO_COLOR_TEMPERATURE_COMMAND) {
+		getColorTemperatureEvent(descMap)
+	} else if (descMap.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER && descMap.commandInt == STEP_COMMAND) {
+		getLevelEvent(descMap)
+	} else if (descMap.clusterInt == zigbee.BASIC_CLUSTER && descMap.commandInt == WRITE_ATTRIBUTES_COMMAND) {
+		getButtonEvent(descMap)
+	} else {
+		log.debug "Unhandled message came in, its description map: ${descMap}"
+	}
+}
+
+private isAttrOrCmdMessage(description) {
+	(description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))
+}
+
+private getOnOffEvent() {
```

Comment:
What if the device was turned on/off while the hub was offline? Is the device itself not sending us the actual current state info?

I'm generally a bit confused about how much we can actually interact with this device. It seems like the answer is "almost not at all".

Reply:
It is sending its current state with no issues. Problem occurs when user uses on/off switch on mobile app, and while we cannot trigger communication with device from hub side, it would cause "missed event issue" from folk's perspective like:
1) Click on/off button on remote - it has sent "off command", _off_ event has been created
2) Current state is _off_, select _on_ command on app in device page. As we cannot interact with device, remote's internal state is still _off_, while in app it's _on_
3) Once again click on/off button on remote - it will send "on command" and _on_ event will be returned, but as device's state in app is already _on_, it would seem that it had done nothing

And yep, there isn't a lot of interaction possible with this device. Basically speaking, it's almost like Z-Wave battery powered sleepy device.